File: bubble_sort.cpp
/*
* @File      :   bubble_sort.cpp
* @Date      :   2021/10/02 22:00:57
* @Author    :   Quiser
* @Email     :   qs0714@foxmail.com
*/
#include <iostream>
#include <ctime>
#include <vector>
using namespace std;
int random(int a, int b)
{
    return rand() % (b - a) + a;
}
void showvec(vector<int> &v)
{
    for (auto &&i : v)
    {
        std::cout << i << " ";
    }
    std::cout << std::endl;
}

void bubble_sort(vector<int> &l)
{
    bool changed = true;
    while (changed)
    {
        changed = false;
        for (int i = 0; i < l.size() - 1; i++)
        {
            if (l[i] > l[i + 1])
            {
                swap(l[i], l[i + 1]);
                changed = true;
            }
        }
    }
}

int main()
{
    srand((int)time(NULL));
    // cout << "Hello World" << endl;
    vector<int> l = {};
    for (int i = 0; i < 100; i++)
    {
        l.push_back(random(0, 100));
    }
    showvec(l);
    bubble_sort(l);
    showvec(l);
    return 0;
}

File: bubble_sort.py
#!/usr/bin/env python
# _*_ coding: utf-8 _*_

# @File    :   bubble_sort.py
# @Author  :   Quiser
# @Email   :   qs0714@foxmail.com
# @Time    :   2021/10/02 21:57:03
from random import randint


def bubble_sort(l):
    changed = True
    while changed:
        changed = False
        for i in range(len(l)-1):
            if l[i] > l[i+1]:
                l[i], l[i+1] = l[i+1], l[i]
                changed = True


def main():
    l = [randint(0, 100) for i in range(100)]
    print(l)
    bubble_sort(l)
    print(l)


if __name__ == '__main__':
    main()
